src/modules/facility-fixed-fee/facility-fixed-fee.controller.ts

```diff
@@ -23,7 +34,55 @@ export class FacilityFixedFeeController {
   @ApiInternalServerErrorResponse({
     description: 'An internal server error has occurred.',
   })
-  async getFixedFeesForFacility(@Param() params: GetFacilityFixedFeeParamsDto): Promise<GetFacilityFixedFeeResponse> {
+  async getFixedFeesForFacility(@Param() params: FacilityFixedFeeParamsDto): Promise<GetFacilityFixedFeeResponse> {
     return await this.facilityFixedFeeService.getFixedFeesForFacility(params.facilityIdentifier);
   }
+
+  @Post('facilities/:facilityIdentifier/fixed-fees')
+  @ApiOperation({
+    summary: 'Create a new fixed fee for a facility.',
+  })
+  @ApiBody({
+    type: CreateFacilityFixedFeeRequestItem,
+    isArray: true,
+  })
+  @ApiCreatedResponse({
+    description: 'The fixed fee has been successfully created.',
+    type: CreateFacilityFixedFeeResponse,
+  })
+  @ApiNotFoundResponse({
+    description: 'The facility was not found.',
+  })
+  @ApiBadRequestResponse({
+    description: 'Bad request.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async createFixedFeeForFacility(
+    @Param() params: FacilityFixedFeeParamsDto,
+    @Body(new ParseArrayPipe({ items: CreateFacilityFixedFeeRequestItem }))
+    newCreateFacilityFixedFeeRequest: CreateFacilityFixedFeeRequest,
+  ): Promise<CreateFacilityFixedFeeResponse> {
+    const facility = await this.facilityService.getFacilityByIdentifier(params.facilityIdentifier);
+
+    if (facility.facilityStageCode !== '07') {
+      throw new BadRequestException('Bad Request', 'Facility needs to be issued before Fee is created');
+    }
+
+    if (facility.facilityOverallStatus !== 'A') {
+      throw new BadRequestException('Bad Request', 'Facility needs to be activated before Fee is created');
```

Comment:
Shall we say "before a fixed fee is created" to be more consistent with the URL we're using?

Reply:
Changed text to "before a fixed fee is created"